dcos_installer/upgrade.py

```diff
@@ -71,6 +71,25 @@
     exit 1
 fi
 
+# Check if the node has node/cluster checks and run them
+if 3dt check; then
+   # command exists
+   if 3dt check node-poststart; then
+      echo "Node checks passed, proceeding with upgrade"
```

Comment:
I'd recommend being silent on success, since that should be the typical case and the user shouldn't have to think about it.